### PR TITLE
feat(UniversalTable): column pinning improvements TASK-1312

### PR DESCRIPTION
**Pull request by magicznyleszek**
### 📣 Summary
Improve multiple pinned columns handling. Allow pinning columns to right side. Improve pinned column styling based on table having horizontal scrollbar. Add reusable `useViewportSize` and `useWindowEvent` hooks. Add extensive story for testing the component, including left/right column pinning, the amount of columns and much more.

### 💭 Notes
Both `useViewportSize` and `useWindowEvent` hooks were copied from `@mantine/hooks`, retaining the original code as closely as possible. We most possibly will use Mantine in near future, and we should replace both with the ones from Mantine package.

Pinned columns have visual shadow to distinct them from other columns (useful when scrolling table horizontally). That shadow is not being added if table has no horizontal scrollbar.

@magicznyleszek this should be used in actions column added in https://github.com/kobotoolbox/kpi/pull/5309 (should be always pinned to right)

### 👀 Preview steps
Best way to test component changes:
1. run storybook locally
2. go to http://localhost:6006/?path=/docs/misc-universaltable--docs
3. 🟢 verify columns pinning works for all possible combinations - story allows pinning one or two columns on both sides
4. 🟢 verify pinned columns have shadow only if table has lots of columns (i.e. if there is horizontal scrollbar)
5. 🟢 verify all the other props of component work as expected

Verify that Organization Members table still works:
1. ℹ️ have multiple different users
2. for one of the users (e.g. "joe"), use http://kf.kobo.local/admin/organizations/organization/ to add multiple users into joe's organization
3. for one of the users (e.g. "sue") set the role to "admin"
4. enable "Multi-members override" for joe's organization
5. enable feature flag `mmosEnabled`
7. navigate to `#/account/organization/members` (as "joe")
8. 🟢 notice that the table works as previously 👌 

Verify that Recent Project Activity table still works:
1. note: this particular table displays mock data right now
2. for a deployed project go to `#/forms/<uid>/settings/activity`
3. 🟢 notice that the table works as previously 👌

### CODE REVIEW

File: jsapp/js/universalTable/universalTable.component.tsx [hunk 1]
```diff
@@ -129,11 +134,24 @@ export default function UniversalTable<DataItem>(
...
   }
 
+  function getCommonColumnStyles(column: Column<DataItem>): CSSProperties {
+    const isPinned = column.getIsPinned();
+    return {
+      left: isPinned === 'left' ? `${column.getStart('left')}px` : undefined,
+      right: isPinned === 'right' ? `${column.getAfter('right')}px` : undefined,
+      width: `${column.getSize()}px`,
+    };
+  }
```
Comment (Akuukis) [inline jsapp/js/universalTable/universalTable.component.tsx:153]: There's a gap between the pinned columns
image

**Review by Akuukis — CHANGES_REQUESTED** [review]
LGTM, tested storybook part.

File: jsapp/js/universalTable/universalTable.component.tsx (hunk 1 above)
Comment (magicznyleszek) [reply jsapp/js/universalTable/universalTable.component.tsx:153]: Thanks, fixed 👍

**Review by Akuukis — APPROVED** [review]
LGTM, tested storybook part only - works and can confirm the bug above was fixed.

File: jsapp/js/universalTable/universalTable.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+}
+
+const meta: Meta<UniversalTablePropsAndCustomArgs> = {
+  title: 'misc/UniversalTable',
+  component: UniversalTable,
+  argTypes: {
+    hasColumnsPinnedLeft: {
+      options: ['none', 'one', 'multiple'],
+      control: {type: 'radio'},
+      description: '_CUSTOM STORY CONTROL_\n\nMakes some of the columns pinned to left. \n\n Note that column pinning happens at the level of column definition, and we use conditionally predefined mock columns in this story.',
```
Comment (RuthShryock) [inline jsapp/js/universalTable/universalTable.stories.tsx:65]:
```suggestion
      description: '_CUSTOM STORY CONTROL_\n\nPins some of the columns to the left. \n\n Note that column pinning happens at the level of column definition, and we use conditionally predefined mock columns in this story.',
```

File: jsapp/js/universalTable/universalTable.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    },
+    hasColumnsPinnedRight: {
+      options: ['none', 'one', 'multiple'],
+      control: {type: 'radio'},
+      description: '_CUSTOM STORY CONTROL_\n\nMakes some of the columns pinned to right.',
```
Comment (RuthShryock) [inline jsapp/js/universalTable/universalTable.stories.tsx:70]:
```suggestion
      description: '_CUSTOM STORY CONTROL_\n\nPins some of the columns to the right.',
```

File: jsapp/js/universalTable/universalTable.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+    },
+    howManyColumns: {
+      options: ['few', 'lots'],
+      control: {type: 'radio'},
+      description: '_CUSTOM STORY CONTROL_\n\nControls how many columns does the table have. Useful for conditionally testing horizontal scroll.',
```
Comment (RuthShryock) [inline jsapp/js/universalTable/universalTable.stories.tsx:75]:
```suggestion
      description: '_CUSTOM STORY CONTROL_\n\nControls how many columns the table has. Useful for conditionally testing horizontal scroll.',
```

**Review by RuthShryock — APPROVED** [review]

**Pull request merged**